provisioning/baremetal_pod.go

```diff
@@ -1125,3 +1126,18 @@ func GetDeploymentState(client appsclientv1.DeploymentsGetter, targetNamespace s
 func DeleteMetal3Deployment(info *ProvisioningInfo) error {
 	return client.IgnoreNotFound(info.Client.AppsV1().Deployments(info.Namespace).Delete(context.Background(), baremetalDeploymentName, metav1.DeleteOptions{}))
 }
+
+func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (string, error) {
+	listOptions := metav1.ListOptions{
+		LabelSelector: metal3AppName,
+		FieldSelector: "status.hostIP",
+	}
+
+	podList, err := podClient.Pods(targetNamespace).List(context.Background(), listOptions)
+	if err == nil {
+		// We expect only one pod with the above LabelSelector
+		hostIP := podList.Items[0].Status.HostIP
```

Comment:
We don't expect more than one, but this will panic if there aren't any.

Reply:
Checking if podList.Items[] is non empty before accessing 1st value.